SwanContents/swancontents/filemanager/eos/fileio.py

```diff
@@ -132,45 +132,46 @@ def read_notebook_from_local(path, encoding='utf-8', as_version=4):
                 return nbformat.v4.new_notebook()
 
         # If the path on the host is defined, save a copy of the notebook there
-        if ('USERDATA_PATH' in os.environ and os.path.isdir(os.environ['USERDATA_PATH'])):
+        if 'USERDATA_PATH' in os.environ and os.path.isdir(os.environ['USERDATA_PATH']):
             # Define the filename for the local copy
             dirname, basename = os.path.split(os_path)
             local_fname = "-".join([os.environ['USER'], "nb", str(int(time.time())), dirname.replace(os.sep, "-"), basename])
             local_path = os.path.join(os.environ['USERDATA_PATH'], local_fname)
 
             # Write the notebook locally and check for consistency
             local_retry = 10
-            write_notebook_to_local(local_path, nb)
-            for i in range(0, local_retry):
-                read_nb = read_notebook_from_local(local_path)
-                if (nb == read_nb):
+            # Use asyncio.to_thread to offload blocking I/O operations
+            await asyncio.to_thread(write_notebook_to_local, local_path, nb)
+            for i in range(local_retry):
+                read_nb = await asyncio.to_thread(read_notebook_from_local, local_path)
+                if nb == read_nb:
                     break
                 else:
-                    #time.sleep(0.1*2**i)    # Backoff on retry (100ms to 51.2s)
-                    time.sleep(0.5)
```

Comment:
Can we safely remove this sleep? Why was it there @diocas ?

Reply:
I'm not sure... Actually, now that we have k8s and no longer mount a local path, we could even remove all of this... The whole method...

 (this was the code we used in the old day to save files locally in puppet nodes, in case EOS was down... This was a measure we introduced to be able to recover data if users complained...)

Reply:
Remove the entire method, but the part that calls the atomic_writing context manager.